### ISSUE TITLE
On Glimmer, components with `{{link-to}}` fail unit tests

### ISSUE THREAD

**Issue by wagenet** [emberjs/ember.js#11150]
Make a component with a `{{link-to}}` in it. Try to unit test it. You'll get a failure due to a missing router. This worked on canary prior to glimmer being merged.

**Comment by bdvholmes** [comment]
I was looking into a similar issue this morning. Although  It wasn't complaining about a missing router, instead about a missing template / view.

Specifically https://github.com/emberjs/ember.js/blob/master/packages/ember-routing-views/lib/initializers/link-to-component.js#L8 was never being called.

**Comment by rwjblue** [comment]
@bdvholmes - That should be registered automatically for unit tests in ember-qunit@0.3.2, but is not registered in `moduleForComponent` based integration tests due to the issue I describe in https://github.com/switchfly/ember-test-helpers/issues/46#issuecomment-100247054.  The solution is outlined there, but I haven't had time to tackle it yet.

**Comment by bdvholmes** [comment]
@rwjblue :+1: awesome thanks, I'll leave you to the missing router error.

**Comment by rwjblue** [comment]
https://github.com/switchfly/ember-test-helpers/pull/49 should address this (will need to verify once merged).

**Comment by nathanhammond** [comment]
This does not appear to have solved the issue. Still having the same test failures using 0.4.0 in our app.

**Comment by rwjblue** [comment]
Yep, we are closer (the router is available now), but it is not initialized properly.  More whack-a-mole to play...

**Comment by csantero** [comment]
Made a repo demonstrating the issue: https://github.com/csantero/link-to-error-demo

**Comment by SirZach** [comment]
Might be redundant information but I'm seeing similar problems.

```
not ok 142 PhantomJS 1.9 - component:game-listing: it renders
    ---
        actual: >
            null
        message: >
            Died on test #2     at http://localhost:7357/assets/test-support.js:2615
                at test (http://localhost:7357/assets/test-support.js:1773)
                at http://localhost:7357/assets/webatrice.js:20234
                at http://localhost:7357/assets/vendor.js:150
                at tryFinally (http://localhost:7357/assets/vendor.js:30)
                at http://localhost:7357/assets/vendor.js:156
                at http://localhost:7357/assets/test-loader.js:29
                at http://localhost:7357/assets/test-loader.js:21
                at http://localhost:7357/assets/test-loader.js:40
                at http://localhost:7357/assets/test-support.js:5381: Assertion Failed: HTMLBars error: Could not find component named "-link-to" (no component or template with that name was found)
        Log: >
```

**Comment by rwjblue** [comment]
@SirZach - That error is not the same as the one this issue is dealing with, later versions of ember-qunit should fix that particular error (but will likely trigger the one that this issue is intended to track/fix).

**Comment by SirZach** [comment]
@rwjblue I am indeed getting a different error after upgrading to 0.3.14.

```
not ok 140 PhantomJS 1.9 - component:game-listing-player: it renders
    ---
        actual: >
            null
        message: >
            Died on test #2     at http://localhost:7357/assets/test-support.js:2852
                at test (http://localhost:7357/assets/test-support.js:1808)
                at http://localhost:7357/assets/webatrice.js:20204
                at http://localhost:7357/assets/vendor.js:150
                at tryFinally (http://localhost:7357/assets/vendor.js:30)
                at http://localhost:7357/assets/vendor.js:156
                at http://localhost:7357/assets/test-loader.js:29
                at http://localhost:7357/assets/test-loader.js:21
                at http://localhost:7357/assets/test-loader.js:40
                at http://localhost:7357/assets/test-support.js:5624: Assertion Failed: Cannot call get with 'targetState' on an undefined object.
        Log: >
```

**Comment by visoft** [comment]
@SirZach Were you able to solve `Cannot call get with 'targetState' on an undefined object.`? It seems to be related to possibly having `link-to` in a component because the error goes away when I strip those from my template.

**Comment by SirZach** [comment]
@visoft I haven't tested with the latest ember-1.13.3 yet. I'll try to remember to update when I get a chance to do so.

**Comment by nathanhammond** [comment]
This was addressed in #11639 and #11644. :)
